### PR TITLE
final_dev

### PR DESCRIPTION
**Pull request by aswath1**
# PR Details
## Description
The detailed design for Stop and Dwell Strategic Plugin was created as part ofTT-15: Create detailed design for Stop Controlled Dwell Time Strategic Plugin
DONE
 . The goal of this story is to do development and add necessary unit test based on the detailed design document, and publish the code to the GitHub.
Then second implementation story could be for generating the maneuver plan in the plan_maneuvers_callback() function.

## Related Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/2072
https://github.com/usdot-fhwa-stol/carma-platform/issues/2071

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [x] I have added any new packages to the sonar-scanner.properties file
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by MishkaMN — CHANGES_REQUESTED** [review]
Overall looks good. Need to add it to the carma/plugins.launch.py

File: carma/launch/plugins.launch.py [hunk 1]
```diff
@@ -519,12 +522,48 @@ def generate_launch_description():
...
+                    ("roadway_objects", [ EnvironmentVariable('CARMA_ENV_NS', default_value=''), "/roadway_objects" ] ),
+                    ("incoming_spat", [ EnvironmentVariable('CARMA_MSG_NS', default_value=''), "/incoming_spat" ] ),
+                    ("plugin_discovery", [ EnvironmentVariable('CARMA_GUIDE_NS', default_value=''), "/plugin_discovery" ] ),
+                    ("route", [ EnvironmentVariable('CARMA_GUIDE_NS', default_value=''), "/route" ] ),
+                    ("maneuver_plan", [ EnvironmentVariable('CARMA_GUIDE_NS', default_value=''), "/final_maneuver_plan" ] ),
+                    ("outgoing_mobility_operation", [ EnvironmentVariable('CARMA_MSG_NS', default_value=''), "/outgoing_mobility_operation" ] ),
+                    ("incoming_mobility_operation", [ EnvironmentVariable('CARMA_MSG_NS', default_value=''), "/incoming_mobility_operation" ] ),
+                    ("bsm_outbound", [ EnvironmentVariable('CARMA_MSG_NS', default_value=''), "/bsm_outbound" ] ),
+                    ("current_pose", [ EnvironmentVariable('CARMA_LOCZ_NS', default_value=''), "/current_pose" ] ),
```
Comment (MishkaMN) [inline carma/launch/plugins.launch.py:550]: Add `("state", [ EnvironmentVariable('CARMA_GUIDE_NS', default_value=''), "/state" ] ),`

**Comment by MishkaMN** [comment]
Looks good to me.

**Comment by maefromm** [comment]
@MishkaMN  can the 2 issues above be closed now?

**Comment by MishkaMN** [comment]
> @MishkaMN  can the 2 issues above be closed now? 

@maefromm yes. They can be closed now

**Pull request merged**